iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorToGPU.cpp

```diff
@@ -91,21 +91,221 @@ static void createAsyncGroups(FuncOp funcOp) {
   }
 }
 
+/// Checks if the offsets, sizes and strides with src, form a no-op
+/// subview. This is true if
+/// 1) The offsets are 0
+/// 2) The strides are 1
+/// 3) The sizes are same as that of the src.
+/// For (3) when the shape is dynamic if the `src` is defined using an operation
+/// that implements the `ShapeAwareOpInterface` (like
+/// `hal.interface.binding.subspan`) then we can use that to check dynamic
+/// equality.
+/// Note: This could be written as a canonicalizer, but the subview formed
+/// when there are dynamic shapes involved will have affine maps
+/// that shouldnt be there. Resolving that is a pain. So dont generate the
+/// subview to begin with.
+static bool generatesNoOpSubView(Value src, ArrayRef<OpFoldResult> offsets,
```

Comment:
I thought that this code was familiar :) . If I am not mistaken this is taken from `LinalgBufferize`.
Some commentary, I hated writing this in the first place. This should really be a separate pattern that kicks in with canonicalize with IREE, but dont know how to do that.
Could you move this into a utility function in `Codegen/Utils`

Reply:
we actually don't need this code, we only care about the rank reducing case so we can skip the code calling it below

Reply:
Makes sense, in that case I wont move the pattern to utils